andela/ah-backend-realers
# #166187852 Users should receive a JWT on successful Login

#### What does this PR do?

Generates a Json web token when a user successfully logs in

#### Description of task completed
* Created an `AuthenticationToken` class that has encoding and decoding methods for JWT's
* Created  a `JWTAuthentication` class in the `backends.py` file
* Wrote tests for JWT authentication functionality

#### How should this be manually tested on your local machine
* Make sure you [postman](https://www.getpostman.com/downloads/) installed on your local machine
* Do a `git fetch origin ft-jwt-auth-166187852`
* Do a `git checkout ft-jwt-auth-166187852`
* Run this command `$ python manage.py runserver`
* The url provided in your terminal should be tested on postman with the prefixes below
 `/api/users/` for `POST` method when the user is registering.

use the format below
`{
       "user":{
              "email": "email",
              "username": "username",
              "password": "password"
         }
               }`
* Then login on postman with the prefixes below
`/api/users/login/` for `POST` method.

use the format below
`{
       "user":{
              "email": "email",
               "password": "password"
         }
               }`
* After logging in, you should have received the token
* Get All the users using postman with the prefixes below
`/api/user/ on a `GET` method. You should be able to get all your details.
* Try it again but with an invalid token


finishes [#166187852](https://www.pivotaltracker.com/n/projects/2350340/stories/166187852)
